src/sql/workbench/contrib/notebook/browser/models/notebookInput.ts

```diff
@@ -420,6 +421,7 @@ export abstract class NotebookInput extends EditorInput implements INotebookInpu
 				}
 			} else {
 				const textEditorModelReference = await this.textModelService.createModelReference(this.resource);
+				this._register(textEditorModelReference);
```

Comment:
We also need to dispose this one, otherwise it will get returned again when opening a file with the same name, but with will have a null object reference that causes the next line to throw an undefined reference error.